server/service/client.go

```diff
@@ -468,6 +468,27 @@ func (c *Client) ApplyGroup(
 	}
 
 	if specs.AppConfig != nil {
+		if _, ok := specs.AppConfig.(map[string]interface{}); ok {
+			// Clear out SSO settings if not provided.
+			if _, ok := specs.AppConfig.(map[string]interface{})["sso_settings"]; !ok {
+				specs.AppConfig.(map[string]interface{})["sso_settings"] = fleet.SSOSettings{}
+			}
+
+			// Clear out SMTP settings if not provided.
+			if _, ok := specs.AppConfig.(map[string]interface{})["smtp_settings"]; !ok {
+				specs.AppConfig.(map[string]interface{})["smtp_settings"] = fleet.SMTPSettings{}
+			}
+
+			// Clear out MDM end_user_authentication settings if not provided.
+			// TODO -- clear out all MDM settings if not provided?
+			if _, ok := specs.AppConfig.(map[string]interface{})["mdm"]; ok {
+				mdm := specs.AppConfig.(map[string]interface{})["mdm"]
+				if _, ok := mdm.(map[string]interface{})["end_user_authentication"]; !ok {
```

Comment:
Same here, If there's no "mdm" key (`mdm` variable will be `nil`) this will panic, right?

Reply:
kk got it, third time's a charm 🤞